components/provisioning/org.wso2.carbon.identity.provisioning.connector.salesforce/src/main/java/org/wso2/carbon/identity/provisioning/connector/salesforce/SalesforceProvisioningConnector.java

```diff
@@ -127,15 +127,17 @@ private JSONObject buildJsonObject(ProvisioningEntity provisioningEntity)
 
         boolean isDebugEnabled = log.isDebugEnabled();
 
-        String provisioningPatternKey = "sf-prov-pattern";
-        String provisioningSeparatorKey = "sf-prov-separator";
-        String idpName_key = "identityProviderName";
-        String userIdClaimUriKey = "userIdClaimUri";
-        String provisioningDomainKey = "sf-prov-domainName";
-
-        String provisioningPattern = this.configHolder.getValue(provisioningPatternKey);
-        String provisioningSeparator = this.configHolder.getValue(provisioningSeparatorKey);
-        String idpName = this.configHolder.getValue(idpName_key);
+        String provisioningPattern = this.configHolder.getValue(SalesforceConnectorConstants.PropertyConfig.PROVISIONING_PATTERN_KEY);
+        if (provisioningPattern == null) {
+            log.info("Provisioning pattern is not defined, hence using default provisioning pattern");
+            provisioningPattern = SalesforceConnectorConstants.PropertyConfig.DEFAULT_PROVISIONING_PATTERN;
+        }
+        String provisioningSeparator = this.configHolder.getValue(SalesforceConnectorConstants.PropertyConfig.PROVISIONING_SEPERATOR_KEY);
+        if (provisioningSeparator == null) {
```

Comment:
Check StringUtils.isBlank() because incase the UI logic changes slightly this backend logic will break. We have to write code as robust as possible.